Add support for openSUSE Leap 15.1, fix build issues

dev-zero opened the pull request:
Unfortunately, the CUDA devs pulled CUDA-8 from the repo, which is why I had to use CUDA-9.2 instead. Furthermore, openSUSE Leap 15.0 would also be supported by this, but the CUDA repo is still not updated :(

dev-zero commented:
I've updated the package to properly build with Opensuse Leap 15.0 since there is now a CUDA toolkit for Leap/SLE 15.0

Keenuts commented on Makefile:
```diff
@@ -134,7 +134,7 @@ LIB_LDLIBS_STATIC  += -l:libtirpc.a
 endif
 ifeq ($(WITH_SECCOMP), yes)
-LIB_CPPFLAGS       += -DWITH_SECCOMP
+LIB_CPPFLAGS       += -DWITH_SECCOMP $(shell pkg-config --silence-errors --cflags libseccomp)
```
--silence-errors ? Do we really want to do that ? Shouldn't we fail here instead of later at link-time ?

dev-zero replied:
Agreed. Unfortunately even without `--silence-errors` will `make` continue, but at least there will be a visible error message.

dev-zero commented:
... and now rebased on current master (1.0.2)

guptaNswati commented:
@dev-zero Can you please sign and email this CLA https://github.com/NVIDIA/libnvidia-container/blob/master/CLA to digits@nvidia.com  cc swatig@nvidia.com before we review and merge your PR. Thanks :)

RenaudWasTaken commented on Makefile:
```diff
@@ -306,7 +306,7 @@ docker-%:
 	$(MKDIR) -p $(DIST_DIR)/$${image/:} ;\
 	$(DOCKER) build --network=host \
-                    --build-arg IMAGESPEC=$* \
+                    --build-arg IMAGESPEC=$${image//_//} \
```
How do you invoke this target?

dev-zero replied:
```console
$ make docker-opensuse_leap:15.1 TAG=1
```
... pretty much the same way as you would for a centos, just that the name is now `opensuse_leap`.

The catch here is that there is more than one openSUSE distribution: [LEAP](https://hub.docker.com/r/opensuse/leap) and [Tumbleweed](https://hub.docker.com/r/opensuse/tumbleweed), which are both in the `opensuse` namespace. It would be preferable to do this directly in `Dockerfile.opensuse_leap` instead, but that doesn't work afaik.

dev-zero replied:
instead of an `_`, we could also use a `-`, which is what openSUSE itself is using in `/etc/os-release`

RenaudWasTaken commented on mk/nvidia-modprobe.mk:
```diff
@@ -41,7 +41,10 @@ $(LIB_SRCS): $(SRCS_DIR)/.download_stamp
 .PHONY: all install clean
 
-all: $(LIB_STATIC)($(LIB_OBJS))
+all: $(LIB_STATIC)
```
Can you explain this why this is needed :) ?

dev-zero replied:
you're basically running into this: https://www.gnu.org/software/make/manual/make.html#Archive-Pitfalls

GNU `make` will run parallel instantiations of `ar` to add each single `.o` to the `.a.`, leading to corruption of the archive.

By replacing `all: $(LIB_STATIC)($(LIB_OBJS))` with

```
all: $(LIB_STATIC)

$(LIB_STATIC): $(LIB_OBJS)
	$(AR) rs $@ $^
```

I turn this into a single call to `ar`, avoiding the race condition.

RenaudWasTaken commented:
We got your CLA! Can you detail a bit your PR? I'll merge after that.

Also super important but non blocking for this PR. We'll be relicensing this repository pretty soon (so you don't have to sign a CLA anymore).

Do you agree if we relicense your contribution to from BSD to Apache2?

dev-zero commented:
> Also super important but non blocking for this PR. We'll be relicensing this repository pretty soon (so you don't have to sign a CLA anymore).
> 
> Do you agree if we relicense your contribution to from BSD to Apache2?

sure, please go ahead

dev-zero commented:
I've rebased everything to recent `master` and renamed the distro from `opensuse_leap` to `opensuse-leap` to match the `ID` in `/etc/os-release`.

The problem fixed with commit 618cf4d is something you likely encounter for all other distros as well as soon as you do `make docker-$WHATEVERDISTRO TAG=1` as someone else than `root`: `/mnt` is usually only writable by `root`.

RenaudWasTaken commented:
Merged in https://github.com/NVIDIA/libnvidia-container/commit/688495e6b51de4496c0f4e09ec456f5e26a9f069

I did this manually as we moved our code to gitlab.com/NVIDIA
These are still your changes that were merged :)

Sorry this took so long, I expect to have time to release the packages next week.
